tsl/test/expected/correlated_constraints.out

```diff
@@ -0,0 +1,285 @@
+-- "sensor_id" column
+SELECT * FROM _timescaledb_catalog.dimension_slice WHERE dimension_id = :dimension_id;
+ id | dimension_id |     range_start      |      range_end      
+----+--------------+----------------------+---------------------
+  3 |            2 | -9223372036854775808 | 9223372036854775807
+  4 |            2 |                    1 |                   9
+(2 rows)
+
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE constraint_name LIKE '_$CC_con%' ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        1 |                  4 | _$CC_constraint_4 | 
+        2 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- check chunk compression status
+SELECT chunk_status
+FROM compressed_chunk_info_view
+WHERE hypertable_name = 'sample_table' AND chunk_name = :'CH_NAME';
+ chunk_status 
+--------------
+            1
+(1 row)
+
+-- A query using a WHERE clause on "sensor_id" column will scan the proper chunk
+-- due to chunk exclusion using correlated constraints ranges calculated above
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+          QUERY PLAN          
+------------------------------
+ Seq Scan on _hyper_1_2_chunk
+   Filter: (sensor_id > 9)
+(2 rows)
+
+-- do update, this will change the status of the chunk
+UPDATE sample_table SET name = 'updated row' WHERE cpu = 21.98 AND temperature = 33.123;
+-- check chunk compression status
+SELECT chunk_status
+FROM compressed_chunk_info_view
+WHERE hypertable_name = 'sample_table' AND chunk_name = :'CH_NAME';
+ chunk_status 
+--------------
+            9
+(1 row)
+
+-- The chunk_constraint should point to the MIN_INT/MAX_INT entry now
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE constraint_name LIKE '_$CC_con%' ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        1 |                  3 | _$CC_constraint_3 | 
+        2 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- A query using a WHERE clause on "sensor_id" column will go back to scanning all the chunks
+-- along with an expensive DECOMPRESS on the first chunk
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+                       QUERY PLAN                        
+---------------------------------------------------------
+ Append
+   ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk
+         Vectorized Filter: (sensor_id > 9)
+         ->  Seq Scan on compress_hyper_2_3_chunk
+   ->  Seq Scan on _hyper_1_1_chunk
+         Filter: (sensor_id > 9)
+   ->  Seq Scan on _hyper_1_2_chunk
+         Filter: (sensor_id > 9)
+(8 rows)
+
+-- recompress the partial chunk
+SELECT compress_chunk(:'CH_NAME');
+WARNING:  no index on "sensor_id" found for correlated constraint on chunk "_hyper_1_1_chunk"
```

Comment:
This WARNING is a bit jarring when compressing.  I guess this is related to the fact that min/max is found using a seqscan on the chunk, which is slightly inefficient. But the user will have no idea. Furthermore, nothing is strictly wrong here, so probably should not be WARNING. I think an info message is more appropriate when defining the correlated constraint (add_dimension?), but not on every recompression.

I realize this message is inherited from adaptive chunking, but then it was used in a different context. I worry that a user would interpret this as something being wrong. It is unclear what the action to take is, if any. If a user isn't expected to do anything about this I think we should avoid displaying this warning.

Also, as I mentioned above that we might be able to get the min/max as output from the compression in the future, so this might not be relevant in that case.

Reply:
I do wonder about the performance implications of the extra seqscan. Maybe better to implement the optimization with getting the min/max during compression sooner rather than later. Not for this PR though, but maybe a follow up?

It could be good to test that compress_chunk() is not significantly slowed down on bigger chunks.

Reply:
@erimatnor typically if the correlated columns are being used in WHERE clauses, then the chances are that an INDEX exists on such a column already.